processing/src/main/java/com/dataspark/masking/MaskingUtil.java

```diff
@@ -0,0 +1,159 @@
+  {
+    if(!maskingOn() || dataSource == null){
+      System.out.println("[ MASKING_UTIL ] will not mask: " + fieldName +" of datasource: " + dataSource);
+      return false;
+    }
+
+    //only do if the datasource is a table ie. true source of data and not a nested query datasource
+    final boolean shouldMask = (dataSource instanceof TableDataSource) && !org.apache.commons.lang.StringUtils.isEmpty(
+        fieldName) && metricsToMask().contains(
+        ((TableDataSource) dataSource).getName().concat(".").concat(fieldName));
+
+    System.out.println("[ MASKING_UTIL ] should mask: " + fieldName +" of datasource: "+ dataSource.getNames() +" = " + shouldMask + " ,DatasourceClass: " + dataSource.getClass());
+    return shouldMask;
+  }
+
+  public static long mask(long value)
+  {
+    return applyPrivacyCensor(extrapolate(value));
+  }
+
+  public static long mask(float value)
+  {
+    return applyPrivacyCensor(extrapolate(value));
+  }
+
+  public static long mask(double value)
+  {
+    return applyPrivacyCensor(extrapolate(value));
+  }
+
+  public static HyperLogLogCollector mask(HyperLogLogCollector hal)
+  {
+    final long privacyThreshold = getPrivacyThreshold();
+    final double originalEstimate = hal.estimateCardinality();
+    final long extrapolatedEstimate = extrapolate(originalEstimate);
+
+    if(extrapolatedEstimate < privacyThreshold) {
+      return createFakeHyperLog(privacyThreshold);
+    }//else return extrapolated number.
+
+    return createFakeHyperLog(extrapolatedEstimate);
+  }
+
+  public static Object doMask(Object value)
+  {
+    if(value instanceof Long) {
+      return mask((Long)value);
+    }else if(value instanceof Double) {
+      return mask((Double)value);
+    }else if(value instanceof Float) {
+      return mask((Float)value);
+    }else if(value instanceof HyperLogLogCollector) {
+      return mask((HyperLogLogCollector) value);
+    }
+    return value;
+  }
+
+  public static HyperLogLogCollector createFakeHyperLog(long numFake)
+  {
+    HyperLogLogCollector collector = HyperLogLogCollector.makeLatestCollector();
+    try {
+      MessageDigest MD5 = MessageDigest.getInstance("MD5");
+      for(long i = 0 ; i < numFake ; i++) {
+        //create fake people :)
+        final String fakeId = "DatasparkFakeId_" + i;
+        collector.add(MD5.digest(fakeId.getBytes()));
+      }
+    } catch (NoSuchAlgorithmException e) {
+      e.printStackTrace();
+    }
+    return collector;
+  }
+
+  //info functions
+  public static long getPrivacyThreshold()
+  {
+    return Long.parseLong(System.getProperty("DATASPARK_PRIVACY_THRESHOLD", "200"));
+  }
+
+  public static double getExtrapolationFactor()
+  {
+    return Double.parseDouble(System.getProperty("DATASPARK_EXTRAPOLATION_FACTOR", "1.35"));
+  }
+
+  public static boolean maskingOn(){
+    return Boolean.parseBoolean(System.getProperty("DATASPARK_DO_MASKING", "true"));
```

Comment:
I think the default value of DATASPARK_DO_MASKING should be FALSE, because I assume if DATASPARK_DO_MASKING is absent, user might not have provided other System Variables.